### DISCUSSION TITLE
Is it possible to load theme from an async api

### DISCUSSION THREAD

**Discussion by zrll12** [mantinedev/mantine#5955] [Q&A]
I was creating a app using Tauri, and I need to load window scales according to the user settings.
The Tauri api returns a Promise, but create_theme doesn't support await, so is there any way to set theme using the async way?

**Comment by rtivital** [comment]
Await your promise before passing the value to createTheme function

**Reply by zrll12** [reply]
How should I modify theme object outside of app

**Reply by rtivital** [reply]
MantineProvider theme property accepts any object, you can modify it any way that is suitable for your application.
